src/dll_hashes.rs

```diff
@@ -0,0 +1,22 @@
+pub fn wintun_dll_hash_sha256() -> std::collections::HashMap<&'static str, &'static str> {
+    [
+        (
+            "wintun/bin/amd64/wintun.dll",
+            "e5da8447dc2c320edc0fc52fa01885c103de8c118481f683643cacc3220dafce",
+        ),
+        (
+            "wintun/bin/arm/wintun.dll",
+            "daad267411ecdc70a0535e274d2c3e9da3d0084bdac7662cb8424dd4a031b4d9",
+        ),
+        (
+            "wintun/bin/arm64/wintun.dll",
+            "f7ba89005544be9d85231a9e0d5f23b2d15b3311667e2dad0debd344918a3f80",
+        ),
+        (
+            "wintun/bin/x86/wintun.dll",
+            "d694fa46ab4cfebcb2632d094c7aa97278eef2f8052438621766d863ae98a931",
+        ),
```

Comment:
Are we sure we want to hard-code the hashes here? Isn't one of the points of dynamically loaded libraries that you can update them without updating the software dependent on them? This will break if users would want to do such a thing.

Isn't the `.dll` signed by Microsoft/Wintun? If so, could we just check whether the signature is valid using their known-good certificates/keys?

Reply:
I hope you can give specific implementation code.
> Isn't the `.dll` signed by Microsoft/Wintun? If so, could we just check whether the signature is valid using their known-good certificates/keys?

We can assume that the compilation result files given by wintun.net is the only binary files that can be trusted.

Reply:
Here the powershell result:
```powershell
PS C:\> $filePath = "C:/windows/system32/wintun.dll"
PS C:\> Get-FileHash -Path $filepath -Algorithm SHA256

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          E5DA8447DC2C320EDC0FC52FA01885C103DE8C118481F683643CACC3220DAFCE       C:\windows\system32\wintun.dll

PS C:\> $signature = Get-AuthenticodeSignature -FilePath $filePath
PS C:\> echo $signature

    Directory: C:\windows\system32

SignerCertificate                         Status                    StatusMessage             Path
-----------------                         ------                    -------------             ----
DF98E075A012ED8C86FBCF14854B8F9555CB3D45  Valid                     Signature verified.       wintun.dll
```